src/google/protobuf/compiler/php/php_generator.cc

```diff
@@ -297,6 +302,34 @@ std::string TypeName(const FieldDescriptor* field) {
   }
 }
 
+std::string PhpTypeName(const FieldDescriptor* field, bool is_descriptor) {
+  if (field->is_repeated()) {
+    return "array|\\Google\\Protobuf\\Internal\\RepeatedField";
```

Comment:
It seems correct to only have RepeatedField for getters if that is always returned. In the case of setters, I would argue that having both is useful, because users will potentially want to take a value from a getter and pass it into a setter.